sdcm/utils/k8s/chaos_mesh.py

```diff
@@ -166,36 +183,70 @@ def wait_until_finished(self):
                 LOGGER.debug("'%s' experiment ended.", self._name)
                 return
             elif status == ExperimentStatus.ERROR:
-                raise PodChaosException(msg="Experiment status error",
-                                        k8s_cluster=self._k8s_cluster,
-                                        podchaos_name=self._name,
-                                        namespace=self._namespace)
+                raise ChaosMeshExperimentException(msg="Experiment status error", experiment=self)
             time.sleep(2)
-        raise PodChaosTimeout(msg="Timeout when waiting for ChaosMesh experiment to complete.",
-                              k8s_cluster=self._k8s_cluster,
-                              podchaos_name=self._name,
-                              namespace=self._namespace)
+        raise ChaosMeshTimeout(msg="Timeout when waiting for ChaosMesh experiment to complete.", experiment=self)
+
+    def describe(self):
+        """Runs kubectl describe experiment resource"""
+        return self._k8s_cluster.kubectl(f"describe {self.CHAOS_KIND} {self._name} -n {self._namespace}").stdout
 
 
-class PodFailureExperiment(PodChaosExperiment):
+class PodFailureExperiment(ChaosMeshExperiment):
     """
     This experiment works by replacing container image with dummy image.
     Then it waits for specified duration and rolls back image config.
     """
+    CHAOS_KIND = "PodChaos"
 
-    def __init__(self, pod: "BasePodContainer", duration: str):
+    def __init__(self, pod: "sdcm.cluster_k8s.BasePodContainer", duration: str):
         """Injects fault into a specified Pod to make the Pod unavailable for a period of time.
 
-        'duration' is str type in k8s notation. E.g. 10s, 5m
+            :param sdcm.cluster_k8s.BasePodContainer pod: affected scylla pod
+            :param str duration: how long pod will be unavailable. str type in k8s notation. E.g. 10s, 5m
         """
         # timeout based on duration + 10 seconds margin
         timeout = time_period_str_to_seconds(duration) + 10
         super().__init__(
             pod=pod, name=f"pod-failure-{pod.name}-{datetime.now().strftime('%d-%H.%M.%S')}", timeout=timeout)
         deep_merge(self._experiment, {
-            "kind": "PodChaos",
             "spec": {
                 "action": "pod-failure",
                 "duration": duration,
             }
         })
+
+
+class MemoryStressExperiment(ChaosMeshExperiment):
```

Comment:
It is K8S- specific, so, I think, it must contain something specific to K8S in it's name.

Reply:
It's a good question, in theory something like memory stress we could use the chaos mesh command line, with the same parameters.

Again we might have different classes, we might implement it in the same class.

For now, we should limit it to only k8s.

We didn't yet tried the commandline in VM to understand of it's a path we want to follow.

Reply:
this class inherits from `ChaosMeshExperiment`, chaos-mesh tool is designed only for K8s and this is simple basic information that must be understood by everyone using SCT. Is there special reason that we need to put `k8s` prefix for such basics?

Reply:
If there an equivalent one targeted for VM, then yes we probably need some prefix.
 
But currently there isn't, so I don't think it's needed at this stage

Reply:
`Experiment` is term from chaos-mesh. When we develop function that works on all backends, we'll need to change naming completely.

Reply:
I think we can leave the name with prefixing for now, if it would be come relevant we could revisit this.